### PR TITLE
Improve node error code checking

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

This PR handles the following missing error code checking:
1. Add a test to verify the error in the creation of the genesis directory.
2. Add logging in case `node.loadParticipationKeys` returns an error
3. Add logging in case `os.Rename` has failed.
4. Remove `fmt.Println(msg)`, as the node reporting output is the log file, and not the console.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -181,7 +181,13 @@ func MakeFull(log logging.Logger, rootDir string, cfg config.Local, phonebookAdd
 	ledgerPathnamePrefix := filepath.Join(genesisDir, config.LedgerFilenamePrefix)
 
 	// create initial ledger, if it doesn't exist
-	os.Mkdir(genesisDir, 0700)
+	err = os.Mkdir(genesisDir, 0700)
+	if err != nil {
```
Comment (algorandskiy) [inline node/node.go:185]: nit: `if err != nil && !os.IsExist(err)` instead of nested ifs

**Review by algorandskiy — COMMENTED** [review]
LGTM

**Comment by algorandskiy** [comment]
> Add a test to verify the error in the creation of the genesis directory.

No test

**Comment by tsachiherman** [comment]
> > Add a test to verify the error in the creation of the genesis directory.
> 
> No test

I meant in-code test ( i.e. check ), not a unit test. I think that adding a unit test for this is a good idea.

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**